src/kernel/network.cpp

```diff
@@ -46,6 +141,8 @@ CryptoKernel::Network::Network(CryptoKernel::Log* log,
 
     running = true;
 
+    std::srand(std::time(0));
```

Comment:
I think we should consider seeding the random number generator with something secure.